src/main/java/streams/cta/io/ImageStream.java

```diff
@@ -26,13 +22,26 @@
     public ImageStream(SourceURL url) {
         super(url);
     }
-
     public ImageStream() {
     }
 
+
+    private class MC {
```

Comment:
some minor comment about these private classes would be nice (although, most of the things are clear at the moment, but for the purity of the project!)